File: C++/equals.cpp
#include <iostream>

bool equals(double x[], double y[], int nx, int ny) {

    if (nx != ny) {
        return false;
    }
    for (int i = 0; i < nx; ++i) {
        if (abs(y[i] - x[i]) > 0.0001) {
            return false;
        }
    }
    return true;
}

File: Methods-Java/Java source-code/equals.java


public class equals {
    public static boolean equals_m(Double[] x, Double[] y) {
        if (x == null || y == null) {
            return !(x == null ^ y == null);
        }
        if (x.length != y.length) {
            return false;
        }
        for (int i = 0; i < x.length; ++i) {
            if (Math.abs(y[i] - x[i]) > 0.0001) {
                return false;
            }
        }
        return true;
    }
}